### PR TITLE
Farm APY calculator button + modal

### PR DESCRIPTION
**Pull request by RichardPK**
- Adding Farm APY calculator button to farm card
- Pull in new uikit version to bring in `CalculateIcon`
- Used `IconButton` to wrap the 'CalculateIcon' - this doesn't necessarily match the designs, but I figure it's better to reuse these components.
- Added two `compoundApiHelpers` to make the calculations for the new ROI modal
- APY modal styling + making use of the calculations returned by `compoundApiHelpers`

### CODE REVIEW

File: src/views/Farms/components/FarmCard/ApyButton.tsx [hunk 1]
```diff
@@ -0,0 +1,45 @@
+import React from 'react'
+import BigNumber from 'bignumber.js'
+import { CalculateIcon, IconButton, useModal } from '@pancakeswap-libs/uikit'
+import ApyCalculatorModal from './ApyCalculatorModal'
+
+export interface TokenAddressesObject {
+  56?: string
+  97?: string
+}
```
Comment (hachiojidev) [inline src/views/Farms/components/FarmCard/ApyButton.tsx:9]: We should be able to use the `Address` type in `src/constants/types.ts`

File: src/views/Farms/components/FarmCard/ApyCalculatorModal.tsx [hunk 2]
```diff
@@ -0,0 +1,132 @@
...
+import styled from 'styled-components'
+import { Modal, Text, LinkExternal, Flex } from '@pancakeswap-libs/uikit'
+import useI18n from 'hooks/useI18n'
+import getLiquidityUrlPathParts from 'utils/getLiquidityUrlPathParts'
+import { calculateCakePerThousand, apyModalRoi } from 'utils/compoundApyHelpers'
+
+export interface TokenAddressesObject {
+  56?: string
+  97?: string
+}
```
Comment (hachiojidev) [inline src/views/Farms/components/FarmCard/ApyCalculatorModal.tsx:12]: Same as above

File: src/views/Farms/components/FarmCard/FarmCard.tsx [hunk 3]
```diff
@@ -133,9 +134,25 @@ const FarmCard: React.FC<FarmCardProps> = ({ farm, removed, cakePrice, bnbPrice,
...
+                  lpLabel={lpLabel}
+                  quoteTokenAdresses={quoteTokenAdresses}
+                  quoteTokenSymbol={quoteTokenSymbol}
+                  tokenAddresses={tokenAddresses}
+                  cakePrice={cakePrice}
+                  apy={farm.apy}
+                />
+              </>
+            ) : (
+              'Loading ...'
```
Comment (hachiojidev) [inline src/views/Farms/components/FarmCard/FarmCard.tsx:153]: Can we use the `<Skeleton />` component here instead.

**Review by hachiojidev — CHANGES_REQUESTED** [review]
Nice work. Just a few minor changes from me.

File: src/utils/compoundApyHelpers.ts [hunk 4]
```diff
@@ -0,0 +1,15 @@
+export const calculateCakePerThousand = (numberOfDays, apy, cakePrice) => {
+  const timesCompounded = 365
+  const apyAsDecimal = apy / 100
+  const daysAsDecimalOfYear = numberOfDays / 365
```
Comment (RabbitDoge) [inline src/utils/compoundApyHelpers.ts:4]: I guess this 365 is the same as timesCompounded ?

Comment (RabbitDoge) [reply src/utils/compoundApyHelpers.ts:4]: PS: Can you add a comment header on each of these function, and add unit tests

**Comment by Chef-Chungus** [comment]
@RichardPK - Looks like there's a little bug. There's now a $ in front of the %. The $ shouldn't be there
image

File: src/views/Farms/components/FarmCard/ApyButton.tsx (hunk 1 above)
Comment (RichardPK) [reply src/views/Farms/components/FarmCard/ApyButton.tsx:9]: Nice one, done 👍

File: src/views/Farms/components/FarmCard/ApyCalculatorModal.tsx (hunk 2 above)
Comment (RichardPK) [reply src/views/Farms/components/FarmCard/ApyCalculatorModal.tsx:12]: Nice one, done 👍

File: src/views/Farms/components/FarmCard/FarmCard.tsx (hunk 3 above)
Comment (RichardPK) [reply src/views/Farms/components/FarmCard/FarmCard.tsx]: Done 👍

**Comment by RichardPK** [comment]
@Chef-Chungus ah yes... that was a stray click of the keyboard. Removed, cheers.

File: src/utils/compoundApyHelpers.ts (hunk 4 above)
Comment (RichardPK) [reply src/utils/compoundApyHelpers.ts]: Done. Writing tests for the functions improved the functions a lot, too. 👍

**Review by hachiojidev — APPROVED** [review]

**Review by Chef-Chungus — APPROVED** [review]

**Pull request merged**